### PR TITLE
Fix typos and add new macros for HID keyboard scancodes.

### PR DESCRIPTION
**Pull request by mondalaci**

### CODE REVIEW

File: LUFA/Drivers/USB/Class/Common/HIDClassCommon.h [hunk 1]
```diff
@@ -330,6 +330,23 @@
 		#define HID_KEYBOARD_SC_RIGHT_SHIFT                       0xE5
 		#define HID_KEYBOARD_SC_RIGHT_ALT                         0xE6
 		#define HID_KEYBOARD_SC_RIGHT_GUI                         0xE7
+		#define HID_KEYBOARD_SC_MEDIA_PLAY                        0xE8
```
Comment (eltang) [inline LUFA/Drivers/USB/Class/Common/HIDClassCommon.h]: These usages aren't listed in the keyboard/keypad page. Are you sure this is the right place for them?

Comment (mondalaci) [inline LUFA/Drivers/USB/Class/Common/HIDClassCommon.h]: @eltang You are right that some of these macros shouldn't belong here. Back in the days I was clueless but now I know better.

According to [USB HID usage tables](http://www.usb.org/developers/hidpage/Hut1_12v2.pdf) the relevant pages are:

* Keyboard / Keymap Page (0x07), starting on page 53
* Consumer Page (0x0C), starting on page 75

The `HID_KEYBOARD_SC_*` macros should cover Keyboard / Keymap Page. It'd be worth to introduce another namespace like `HID_KEYBOARD_MEDIA_*` to cover the consumer page.

Does it seem right, @abcminiuser?

I'm about to implement this in the [UHK firmware](https://github.com/UltimateHackingKeyboard/firmware) and backport it to LUFA. No ETA on it yet but I plan to do it within several weeks.

Comment (abcminiuser) [inline LUFA/Drivers/USB/Class/Common/HIDClassCommon.h]: Sounds like a plan. We can never hope to cover all the usage pages/usages, but I think there's a well-defined sub-set of them that most people actually want (i.e. the ones on a typical full function keyboard with media extensions). A new namespace would be a good idea -- migration won't be difficult for end-users.

Comment (mondalaci) [inline LUFA/Drivers/USB/Class/Common/HIDClassCommon.h]: Totally agreed.

Comment (eltang) [inline LUFA/Drivers/USB/Class/Common/HIDClassCommon.h]: How about `HID_CONSUMER_MEDIA_*` or something similar instead? The consumer page contains more than just usages for media control, and a device which uses that page is not necessarily a keyboard.

Comment (mondalaci) [inline LUFA/Drivers/USB/Class/Common/HIDClassCommon.h]: @eltang Works for me.

Comment (NicoHood) [reply LUFA/Drivers/USB/Class/Common/HIDClassCommon.h]: I just stumbled across that code wondering why those definitions are there. They are wrong and do not belong to this file. The media keys have different values. If you declare those definitions "Scancodes - SC" it is wrong. We should revert this change.

Comment (mondalaci) [reply LUFA/Drivers/USB/Class/Common/HIDClassCommon.h]: Admittedly, the consumer page should be used for media key codes, but they aren't completely wrong. Back in the days, I didn't know about the consumer page and only used the keyboard / keypad page. These values worked for me for Linux, but later I realized that they don't work for other OSes. Given that they work on Linux, I think they're not futile. Maybe it should be emphasized that they only work for Linux.

Comment (NicoHood) [reply LUFA/Drivers/USB/Class/Common/HIDClassCommon.h]: If you use those scancodes under linux they work as media keys!?!? They re not even documented. I need to test it, that would be wonderful, but still weird.

Comment (mondalaci) [reply LUFA/Drivers/USB/Class/Common/HIDClassCommon.h]: Yes, the last time they did work on Linux. They're not documented, indeed.

Comment (NicoHood) [reply LUFA/Drivers/USB/Class/Common/HIDClassCommon.h]: It seems to work. I am really surprised. This is ultra awesome!

**Pull request merged**